diff --git a/include/linux/of_graph.h b/include/linux/of_graph.h
index 2b233db..befef42 100644
--- a/include/linux/of_graph.h
+++ b/include/linux/of_graph.h
@@ -37,7 +37,7 @@ struct device_node *of_graph_get_remote_port(const struct device_node *node);
 #else
 
 static inline int of_graph_parse_endpoint(const struct device_node *node,
-					struct of_endpoint *endpoint);
+					struct of_endpoint *endpoint)
 {
 	return -ENOSYS;
 }
