diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -318,7 +318,7 @@ static unsigned int ir_establish_scancod
  * This routine is used to handle evdev EVIOCSKEY ioctl.
  */
 static int ir_setkeycode(struct input_dev *idev,
-			 unsigned int scancode, unsigned int keycode)
+			 int scancode, int keycode)
 {
 	struct rc_dev *rdev = input_get_drvdata(idev);
 	struct ir_scancode_table *rc_tab = &rdev->rc_tab;
@@ -423,7 +423,7 @@ static unsigned int ir_lookup_by_scancod
  * This routine is used to handle evdev EVIOCGKEY ioctl.
  */
 static int ir_getkeycode(struct input_dev *idev,
-			 unsigned int scancode, unsigned int *keycode)
+			 int scancode, int *keycode)
 {
 	struct rc_dev *rdev = input_get_drvdata(idev);
 	struct ir_scancode_table *rc_tab = &rdev->rc_tab;
