--- linux.orig/drivers/media/usb/uvc/uvc_video.c	2012-11-14 14:33:51.995088200 -0500
+++ linux/drivers/media/usb/uvc/uvc_video.c	2012-11-14 14:36:23.510088227 -0500
@@ -1447,7 +1447,7 @@
 
 	switch (dev->speed) {
 	case USB_SPEED_SUPER:
-		return ep->ss_ep_comp.wBytesPerInterval;
+		return ep->ss_ep_comp->desc.wBytesPerInterval;
 	case USB_SPEED_HIGH:
 		psize = usb_endpoint_maxp(&ep->desc);
 		return (psize & 0x07ff) * (1 + ((psize >> 11) & 3));
