diff --git a/drivers/media/pci/ivtv/ivtvfb.c b/drivers/media/pci/ivtv/ivtvfb.c
index 4cb365d..eed697a1 100644
--- a/drivers/media/pci/ivtv/ivtvfb.c
+++ b/drivers/media/pci/ivtv/ivtvfb.c
@@ -1266,11 +1266,13 @@ static int __init ivtvfb_init(void)
 	int err;
 
 #ifdef CONFIG_X86_64
-	if (WARN(pat_enabled(),
+#ifdef CONFIG_X86_PAT
+	if (WARN(pat_enabled,
 		 "ivtvfb needs PAT disabled, boot with nopat kernel parameter\n")) {
 		return -ENODEV;
 	}
 #endif
+#endif
 
 	if (ivtvfb_card_id < -1 || ivtvfb_card_id >= IVTV_MAX_CARDS) {
 		printk(KERN_ERR "ivtvfb:  ivtvfb_card_id parameter is out of range (valid range: -1 - %d)\n",
