diff -r 128af39ba902 drivers/media/video/bt8xx/bttv-input.c
--- a/drivers/media/video/bt8xx/bttv-input.c	Tue Oct 26 13:43:12 2010 -0200
+++ b/drivers/media/video/bt8xx/bttv-input.c	Tue Oct 26 14:17:09 2010 -0200
@@ -303,7 +303,7 @@
 		 * That's why we probe 0x1a (~0x34) first. CB
 		 */
 
-		i2c_new_probed_device(&btv->c.i2c_adap, &info, addr_list, NULL);
+		i2c_new_probed_device(&btv->c.i2c_adap, &info, addr_list);
 		return;
 	}
 
diff -r 128af39ba902 drivers/media/video/cx18/cx18-i2c.c
--- a/drivers/media/video/cx18/cx18-i2c.c	Tue Oct 26 13:43:12 2010 -0200
+++ b/drivers/media/video/cx18/cx18-i2c.c	Tue Oct 26 14:17:09 2010 -0200
@@ -104,7 +104,7 @@
 		break;
 	}
 
-	return i2c_new_probed_device(adap, &info, addr_list, NULL) == NULL ?
+	return i2c_new_probed_device(adap, &info, addr_list) == NULL ?
 	       -1 : 0;
 }
 
diff -r 128af39ba902 drivers/media/video/cx23885/cx23885-i2c.c
--- a/drivers/media/video/cx23885/cx23885-i2c.c	Tue Oct 26 13:43:12 2010 -0200
+++ b/drivers/media/video/cx23885/cx23885-i2c.c	Tue Oct 26 14:17:09 2010 -0200
@@ -354,7 +354,8 @@
 	} else
 		printk(KERN_WARNING "%s: i2c bus %d register FAILED\n",
 			dev->name, bus->nr);
-
+#if 0
+	/* This probe function won't work on older devices */
 	/* Instantiate the IR receiver device, if present */
 	if (0 == bus->i2c_rc) {
 		struct i2c_board_info info;
@@ -369,7 +370,7 @@
 		i2c_new_probed_device(&bus->i2c_adap, &info, addr_list,
 				      i2c_probe_func_quick_read);
 	}
-
+#endif
 	return bus->i2c_rc;
 }
 
diff -r 128af39ba902 drivers/media/video/em28xx/em28xx-cards.c
--- a/drivers/media/video/em28xx/em28xx-cards.c	Tue Oct 26 13:43:12 2010 -0200
+++ b/drivers/media/video/em28xx/em28xx-cards.c	Tue Oct 26 14:17:09 2010 -0200
@@ -2416,7 +2416,7 @@
 
 	if (dev->init_data.name)
 		info.platform_data = &dev->init_data;
-	i2c_new_probed_device(&dev->i2c_adap, &info, addr_list, NULL);
+	i2c_new_probed_device(&dev->i2c_adap, &info, addr_list);
 }
 
 void em28xx_card_setup(struct em28xx *dev)
diff -r 128af39ba902 drivers/media/video/ivtv/ivtv-i2c.c
--- a/drivers/media/video/ivtv/ivtv-i2c.c	Tue Oct 26 13:43:12 2010 -0200
+++ b/drivers/media/video/ivtv/ivtv-i2c.c	Tue Oct 26 14:17:09 2010 -0200
@@ -158,7 +158,7 @@
 			return -1;
 		memset(&info, 0, sizeof(struct i2c_board_info));
 		strlcpy(info.type, type, I2C_NAME_SIZE);
-		return i2c_new_probed_device(adap, &info, addr_list, NULL)
+		return i2c_new_probed_device(adap, &info, addr_list)
 							   == NULL ? -1 : 0;
 	}
 
diff -r 128af39ba902 drivers/media/video/v4l2-common.c
--- a/drivers/media/video/v4l2-common.c	Tue Oct 26 13:43:12 2010 -0200
+++ b/drivers/media/video/v4l2-common.c	Tue Oct 26 14:17:09 2010 -0200
@@ -383,8 +383,7 @@
 
 	/* Create the i2c client */
 	if (info->addr == 0 && probe_addrs)
-		client = i2c_new_probed_device(adapter, info, probe_addrs,
-					       NULL);
+		client = i2c_new_probed_device(adapter, info, probe_addrs);
 	else
 		client = i2c_new_device(adapter, info);
 
